### ISSUE TITLE
Concern on REG reproducibility and potential results mismatch

### ISSUE THREAD

**Issue by linYDTHU** [Martinser/REG#10]
Hi authors,

Thank you for releasing the code and the paper.
We have been carefully trying to reproduce the REG results reported in the paper, and would like to raise a **reproducibility and evaluation concern**, particularly regarding the **REG performance at early training stages and the distinction between CFG and non-CFG evaluation**.

Below we provide concrete commands, logs, and observations.

## 1. Training setup (following the released code)

We trained **SiT-XL/2 + REG** for **100k steps**, following the default configuration as closely as possible.

### Training command

```bash
accelerate launch train.py \
  --report-to=wandb \
  --allow-tf32 \
  --mixed-precision=fp16 \
  --seed=0 \
  --path-type=linear \
  --prediction=v \
  --weighting=uniform \
  --model=SiT-XL/2 \
  --enc-type=dinov2-vit-b \
  --proj-coeff=0.5 \
  --encoder-depth=8 \
  --output-dir=reg_exps \
  --exp-name=linear-dinov2-b-enc8 \
  --batch-size=256 \
  --data-dir=data \
  --cls=0.03 \
  --max-train-steps=100000
```

### Training behavior

The training is stable and convergent. Loss terms behave as expected:

* `loss_final` decreases smoothly from ~0.54 → ~0.43
* `proj_loss` steadily decreases (≈ −0.63 → −0.74)
* No divergence or instability observed

(Full logs available if needed.)

---

## 2. Evaluation protocol

We evaluate the **100k checkpoint** using the provided `generate.py`.

### Evaluation command

```bash
torchrun --nproc_per_node=4 generate.py \
  --model SiT-XL/2 \
  --num-fid-samples 50000 \
  --ckpt reg_exps/linear-dinov2-b-enc8/checkpoints/0100000.pt \
  --path-type=linear \
  --encoder-depth=8 \
  --projector-embed-dims=768 \
  --per-proc-batch-size=64 \
  --mode=sde \
  --num-steps=250 \
  --cfg-scale=1.0 \
  --cls-cfg-scale=1.0 \
  --guidance-high=1.0 \
  --sample-dir samples \
  --cls=768
```

---

## 3. Observed results

* **FID @ 100k steps ≈ 28.23
* This is **far from the ≈8 FID** shown around 100k iterations in **Figure 2** of the paper.
* The discrepancy is more than **3×**

---

## 4. Additional observation: CFG vs non-CFG inconsistency

We further notice a potential **evaluation mismatch** in the paper:

* **Table 1 (non-CFG)** reports extremely low FID values (e.g., REG ≈ 1.8 at 4M iterations).
* **Table 2 (CFG)** reports FID values numerically close to REPA, despite Table 1 suggesting a very strong unconditional model.

However, in our reproduction:

* Strict non-CFG evaluation (no effective guidance) produces **significantly worse FID**.
* The reported non-CFG numbers appear unusually low compared to known ImageNet-256 baselines.

This raises the possibility that **CFG may not have been fully disabled** during the reported “non-CFG” evaluations (e.g., conditional branch still active with scale = 1.0).

---

## 5. Requests for clarification

To help resolve this and ensure fair comparison, could you please clarify:

1. Whether **CFG is fully disabled** during non-CFG evaluation:

   * Is the conditional branch completely removed?
   * Or is `cfg_scale = 1.0` used?

2. Could you **release**:

   * The **100k-step checkpoint** used for Figure 2
   * The **exact evaluation script / flags** used to compute Table 1 non-CFG FID

We believe this clarification would greatly help the community understand and fairly reproduce the reported gains of REG.

Thanks again for the release and looking forward to your response.

Best regards,
Donglin

**Comment by Martinser** [comment]
Hi，您好
方便添加我的微信吗？以便进一步沟通交流？
微信号：wg1158848

Donglin Yang ***@***.***> 于2026年1月15日周四 21:33写道：

**Comment by linYDTHU** [comment]
We would like to thank the authors' help and provide a clarification and update regarding our previous reproduction attempt.

After further investigation, we found that the discrepancy in our earlier results was caused by a **conflict introduced by a local modification to the loss formulation** in our training code. After reverting this change and strictly following the released implementation, we are able to **successfully reproduce the 100k-step REG result**.

Specifically:

* At **100k training steps**, we obtain **FID = 8.40**
* The paper reports **FID = 8.90** at the same stage
* The reproduced result is therefore **consistent with the reported performance**, within expected variance

This confirms that **REG can indeed achieve the reported early-stage performance when the training objective is implemented exactly as in the released code**.

**Comment by Crane-YU** [comment]
@linYDTHU May I know how long does the training time take to reach 100k training steps?
